src/main/java/io/openliberty/tools/common/plugins/util/PrepareFeatureUtil.java

```diff
@@ -0,0 +1,416 @@
+	 * repository
+	 * 
+	 * @param fileFromRepo The file from the repository
+	 * @param groupId      The groupdId
+	 * @param artifactId   The artifactId
+	 * @param fileType     The fileType
+	 * @param version      The version
+	 * @return The repository location of the fileFromRepo file
+	 */
+	private String parseRepositoryLocation(File fileFromRepo, String groupId, String artifactId, String fileType,
+			String version) {
+		String absFileFromRepo = fileFromRepo.getAbsolutePath();
+		groupId = groupId.replace(".", "/");
+		String fileSubString = String.format("%s/%s/%s/%s-%s.%s", groupId, artifactId, version, artifactId, version,
+				fileType);
+
+		return absFileFromRepo.replace(fileSubString, "");
+	}
+
+	/**
+	 * Generate JSON at targetJsonFile location for the list of ESAs in esaFiles
+	 * under the given groupId
+	 * 
+	 * @param targetJsonFile The target file location for the resulting JSON
+	 * @param esaFileMap     Map of esa Files to their groupIds
+	 * @throws PluginExecutionException Throws an error if unable to generate JSON
+	 */
+	public File generateJson(String targetJsonFile, Map<File, String> esaFileMap) throws PluginExecutionException {
+		try {
+			Path targetDir = Files.createTempDirectory("generatedJson");
+			URL installJarURL = null;
+			try {
+				installJarURL = installJarFile.toURI().toURL();
+			} catch (MalformedURLException e) {
+				throw new PluginExecutionException("Could not resolve URL from file " + installJarFile, e);
+			}
+			Map<String, Object> mapBasedInstallKernel = null;
+			File json = null;
+			List<File> esaFileList = new ArrayList<File>();
+			esaFileList.addAll(esaFileMap.keySet());
+			try (final URLClassLoader loader = new URLClassLoader(new URL[] { installJarURL },
+					getClass().getClassLoader())) {
+				mapBasedInstallKernel = createMapBasedInstallKernelInstance(loader, installDirectory);
+				mapBasedInstallKernel.put("individual.esas", esaFileList);
+				mapBasedInstallKernel.put("target.json.dir", targetDir.toFile());
+				mapBasedInstallKernel.put("generate.json.group.id.map", esaFileMap);
+				mapBasedInstallKernel.put("generate.json", true);
+				json = (File) mapBasedInstallKernel.get("generate.json");
+
+				if (mapBasedInstallKernel.get("action.error.message") != null) {
+					debug("generateJson action.exception.stacktrace: "
+							+ mapBasedInstallKernel.get("action.error.stacktrace"));
+				}
+
+			} catch (PrivilegedActionException e) {
+				throw new PluginExecutionException("Could not load the jar " + installJarFile.getAbsolutePath(), e);
+			}
+			File targetFile = new File(targetJsonFile);
+			FileInputStream instream = new FileInputStream(json);
+			targetFile.getParentFile().mkdirs();
+			FileOutputStream outstream = new FileOutputStream(targetFile);
+			byte[] buffer = new byte[1024];
+			int length;
+			while ((length = instream.read(buffer)) > 0) {
+				outstream.write(buffer, 0, length);
+			}
+			instream.close();
+			outstream.close();
```

Comment:
The closing of the streams should probably be in a `final` block after the `catch`.